.github/workflows/publish-helm-chart.yml

```diff
@@ -74,6 +80,64 @@ jobs:
           - name: Helm Chart Linter
             run: |
               helm lint ${{matrix.chartpath}}
+            
+          - name: SEMVER Check
+            # Checks Chart Version Number against SEMVER regex
+            run: | 
+              version=$(helm show chart ${{matrix.chartpath}} | 
+              awk '/version:/ {print $2}')
+
+              # Checks for semantic versioning specification, https://semver.org/
+              semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\
+              (-(0|[1-9A-Za-z-][0-9A-Za-z-]*)(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]\
+              +(\.[0-9A-Za-z-]+)*)?$"
+
+              if [[ $version =~ $semver_regex ]]; then
+                echo "$version uses SEMVER, https://semver.org/"
+              else
+                echo "$version is not using SEMVER, https://semver.org/"
+                exit 1
+              fi 
+
+          - name: Helm Chart Version Check
+            env: 
+              BASE: ${{needs.build-chart-list.outputs.BASE}}
+            run: |
+              # old_chart_version diffs commit that triggered workflow against previous 
+              # commit. Pulls out version number of the existing chart or "old chart".
+              # The git diff gives spits out both version numbers, head -2 | tail -1
+              # Grabs the value on the second line. 
+              old_chart_version=$(git diff "$BASE" ${{ github.sha }} -- ${{matrix.chartpath}} | 
+                                  awk '/version:/ {print $2}' | 
+                                  head -1 | 
+                                  tail -1 | 
+                                  tr -d '"')
+
+              # Grabs the chart version number out of the chart that needs
+              # to be checked. 
+              new_chart_version=$(helm show chart ${{matrix.chartpath}} | 
+              awk '/version:/ {print $2}')
+
+              # Checks if the new chart version number
+              # is > than the existing chart.
+              sortversion() {
+                  [  "$1" = "`echo -e "$1\n$2" | sort -V | head -n1`" ]
+              }
+
+              versioncheck() {
+                  [ "$1" = "$2" ] && return 1 || sortversion $1 $2
+              }
```

Comment:
You may as well combine this into one function! Or directly use it as the conditions of the if block below